src/apps/file/api/file.py

```diff
@@ -98,27 +116,42 @@ async def answer_download(
 
 
 async def check_file_uploaded(
+    applet_id: uuid.UUID,
     schema: FileCheckRequest,
     user: User = Depends(get_current_user),
+    session: AsyncSession = Depends(get_session),
 ) -> ResponseMulti[FileExistenceResponse]:
     """Provides the information if the files is uploaded."""
 
+    if not await UserAppletAccessCRUD(session).get_by_roles(
+        user.id,
+        applet_id,
+        [Role.OWNER, Role.MANAGER, Role.REVIEWER],
+    ):
+        raise AnswerViewAccessDenied()
+
     cdn_client = CDNClient(settings.cdn, env=settings.env)
```

Comment:
Do you use bucket for answers here?

Reply:
Arbitrary server should be supported here. You upload files to arbitrary servers as well